ethereum/contracts/ChannelAccess.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.7.6;
+pragma experimental ABIEncoderV2;
+
+abstract contract ChannelAccess {
```

Comment:
Can we get a high level comment describing the problem being solved by this as a guide to readers/auditors?

Reply:
added docs in latest upload.